scripts/teamDataScript.js

```diff
@@ -319,21 +319,23 @@ function loadMatch()
     var matchInputHTML = `<input type="number" class="matchNumInput match-frequency js_clear_on_load">`;
     var matchRow = `<tr> <td class="match-element scoutingTableRow">Match #</td> <td class="scoutingTableRow">${matchInputHTML}</td> </tr>`;
     $('#matchQuestionTable').append(sanitizeHTMLLine(matchRow)); //default for all teams: row for inputting what match it was
-    for (var i = 0; i < cookieList.length; i++)
-    {
-        var currCName = cookieList[i].trim().split('=')[0];
-
-        if (!(currCName.startsWith(cname))) continue;
 
-        var element = currCName.substring(currCName.lastIndexOf('/')+1);
-        var value = getCookie(currCName);
+    //retrieve elements for current season
+    var elementsObj = JSON.parse(getCookie(season_config_cookie_name(season)));
+    var elementKeys = Object.keys(elementsObj);
+    for (var i = 0; i < elementKeys.length; i++)
+    {
+        var element = elementKeys[i];
+        var nestedObject = elementsObj[element]; //input type is in a nested object
+        var inputType = nestedObject['inputType'];
 
-        var inputHTML = `<input type="number" class="match-frequency js_clear_on_load">`;
+        var inputHTML = `<input type="${(inputType==='numeric')?'number':'text'}" class="match-frequency js_clear_on_load">`;
```

Comment:
we might want to use `textarea` in place of the `text` and setup the auto sizing.